js/tji-datasets-explore.js

```diff
@@ -307,64 +309,124 @@ TJIChartView.prototype.transform_data = function() {
-    _.each(f.values, function(v) {
-      var input = jQuery('<input/>', {
-        type: "checkbox",
-        checked: "checked",
-        id: v,
-        name: f.key,
-        value: v,
-      });
-      var label = jQuery('<label/>', {
-        for: v,
-      }).text(v);
-      fieldset.append(jQuery('<div></div>').append(input, label));
-    });
+  _.each(that.filter_configs, function(filter_config) {
+    var fieldset;
+    switch(filter_config.type) {
+      case 'autocomplete': 
+        fieldset = that.create_filter_autocomplete(filter_config);
+        break;
+      case 'checkbox':
+      default:
+        fieldset = that.create_filter_checkboxes(filter_config);
+        break;
+    }
     fieldsets.push(fieldset);
   });
   
-  this.$form = jQuery('<form />').append(fieldsets);
+  this.$form = jQuery('<form />', {
+    class: 'tji-chart-filters',
+  }).append(fieldsets);
   this.state.active_filters = this.$form.serializeArray();  
-  jQuery(this.filters_elt_id).append(this.$form);
+  jQuery(this.filters_elt_selector).append(this.$form);
   
 }
 
+TJIChartView.prototype.create_filter_checkboxes = function(filter) {
+  var that = this;
+  var fieldset = jQuery('<fieldset><legend>' + filter.name.replace(/_/g, " ") + '<i class="fas fa-caret-down"></i></legend></fieldset>');
+  _.each(filter.values, function(v) {
+    fieldset.append(that.create_filter_checkbox(filter.name, v));
+  });
+  return fieldset;
+}
+
+TJIChartView.prototype.create_filter_checkbox = function(name, value) {
+  var input = jQuery('<input/>', {
+      class: "tji-chart-filters__checkbox",
+      type: "checkbox",
+      checked: "checked",
+      id: 'TJIChartView__filter-' + value,
+      name: name,
+      value: value,
+    });
+  var label = jQuery('<label/>', {
+    for: 'TJIChartView__filter-' + value,
+  }).text(value);
+  var container = jQuery('<div/>', {
+      class: "tji-chart-filters__filter",
+  }).append(input, label);
+  return container;
+}
+
+TJIChartView.prototype.create_filter_autocomplete = function(filter) {
+  var that = this;
+  var fieldset = jQuery('<fieldset><legend>' + filter.name.replace(/_/g, " ") + '<i class="fas fa-caret-down"></i></legend></fieldset>');
+  var input = jQuery('<input/>', {
+    class: "tji-chart-filters__text",
+    type: "text",
+  });
+  fieldset.append(jQuery('<div/>',{
+    class: "tji-chart-filters__autocomplete",
+  }).append(input));
+  var $auto_complete = jQuery('<div />', {
+    class: "tji-chart-filters__autocomplete-list"
+  }).insertAfter(input);
+  var auto_complete = new autoComplete({
+      selector: input[0],
+      minChars: 1,
+      delay: 150, 
+      source: function(term, suggest){
+          term = term.toLowerCase();
+          suggest(_.filter(filter.values, function(v){
+            return ~v.toLowerCase().indexOf(term);
+          }));
+      },
+      onSelect: function(event, term, item) {
```

Comment:
If I just type `TRAVIS` and hit enter, nothing happens. I have to click on the autocomplete-suggested `TRAVIS`, or arrow-down and then hit enter to select it.

![screen shot 2018-07-08 at 2 29 40 pm](https://user-images.githubusercontent.com/1421476/42423172-8d1570ba-82bb-11e8-8503-90245ae82627.png)

Reply:
I'll take a minute to investigate this. Since we're using an external library, that has control of event capture, it might be something we're stuck with -- tho, if I come up with something, yay for contributing to OS projects.

Reply:
@wetchler I took more than a minute, and now you're hungry, but hey, it's implemented!